lib/anno_utils.pm

```diff
@@ -0,0 +1,2917 @@
+    #
+    my $empty_input_msg = ("ERROR:Missing required inputs--must specify at least one genome \n".
+            "and/or a string of genome names separated by ';', '\n' or '|' (without quotes).\n");
+    Bio::KBase::Exceptions::ArgumentValidationError->throw(
+        error        => $empty_input_msg,
+        method_name  => 'bulk_rast_genomes'
+    ) unless ref $bulk_inparams eq 'ARRAY' && @{$bulk_inparams};
+
+    my $anngns = [];
+    foreach my $parm (@{$bulk_inparams}) {
+        my $rast_out = $self->rast_genome($parm);
+        if (keys %{ $rast_out }) {
+            push (@$anngns, $rast_out->{output_genome_ref});
+        }
+    }
+
+    # create, save and then return that GenomeSet object's ref
+    my $kbutil = new installed_clients::kb_SetUtilitiesClient($self->{call_back_url});
+    my $kbutil_output = $kbutil->KButil_Build_GenomeSet({
+        workspace_name => $ws,
+        input_refs => $anngns,
+        output_name => $out_genomeSet,
+        desc => 'GenmeSet generated from RAST annotated genomes/assemblies'
+    });
+
+    return {"output_genomeSet_ref"=>$ws."/".$out_genomeSet,
+            "output_workspace"=>$ws,
+            "report_name"=>$kbutil_output->{report_name},
+            "report_ref"=>$kbutil_output->{report_ref}};
+}
+
+#
+## for use with arrays, e.g.,
+## my @unique = uniq(@data);
+#
+sub _uniq {
+    my $self = shift;
+    keys { map { $_ => 1 } @_ };
+}
+
+#
+## for use with array refs, e.g.,
+## my $unique = uniq($data);
+#
+sub _uniq_ref {
+    my $self = shift;
+    [ keys { map { $_ => 1 } @{$_[0]} } ];
+}
+
+#
+## for use to check if a key value is in an array of hash
+#
+sub _value_in_array {
+     my ($self, $val, $arr) = @_;
+     for my $href (@{$arr}) {
+        if ($href->{object_ref} eq $val) {
+            return 1;
+        }
+     }
+     return 0;
+}
+
+#
+## Initialization
+#
+sub doInitialization {
+    my $self = shift;
+
+    $self->{_token} = $self->{ctx}->token();
+    $self->{_username} = $self->{ctx}->user_id();
+    $self->{_method} = $self->{ctx}->method();
+    $self->{_provenance} = $self->{ctx}->provenance();
+
+    $self->{ws_url} = $self->{config}->{'workspace-url'};
+    $self->{call_back_url} = $ENV{ SDK_CALLBACK_URL };
+    $self->{rast_scratch} = $self->{config}->{'scratch'};
+    $self->{genome_dir} = $self->_create_rast_subdir($self->{rast_scratch},
+                                                     "genome_annotation_dir_");
+
+    $self->{max_contigs} = 10000;
+
+    die "no workspace-url defined" unless $self->{ws_url};
+
+    $self->{ws_client} = new installed_clients::WorkspaceClient(
+                             $self->{ws_url}, token => $self->{_token});
+
+    return 1;
+}
+
+#
+## constructor
+#
+sub new {
+    my $class = shift;
+
+    my $self = {
+        'config' => shift,
+        'ctx' => shift
+    };
+
+    bless $self, $class;
+    $self->doInitialization();
+
+    return $self;             # Return the reference to the hash.
+}
```

Comment:
put the constructor at the top of the file

Reply:
This is a preference issue.  I prefer that to be at the bottom of the file.